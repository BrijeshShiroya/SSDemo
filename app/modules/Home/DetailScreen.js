import { Container, Content, Text } from 'native-base';
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'react-native';
import { useSelector } from 'react-redux';
import {
  AppBackground,
  CustomButton,
  CustomHeader,
  CustomSlider
} from '../../components';
import { AppConstants, Strings } from '../../constants';
import { Colors, Icons } from '../../theme';
import styles from './styles/DetailScreenStyles';

const DetailScreen = ({ navigation }) => {
  const { user } = useSelector((state) => state.welcome);
  const [slider, setSlider] = useState(0);
  const onVariationPress = (type) => {
    Alert.alert(`You pressed variation ${type}`);
  };
  const renderButtons = () => {
    return (
      <>
        <Text style={styles.btnContainerTitle}>{Strings.variationTitle}</Text>
        <CustomButton
          title={Strings.btnV1}
          style={[styles.btnVariation, { backgroundColor: Colors.transparent }]}
          onPress={() => onVariationPress(1)}
        />
        <CustomButton
          title={Strings.btnV2}
          style={[
            styles.btnVariation,
            { backgroundColor: Colors.btnOrangeNeutral }
          ]}
          onPress={() => onVariationPress(2)}
        />
        <CustomButton
          title={Strings.btnV3}
          style={[styles.btnVariation, { backgroundColor: Colors.btnOrange }]}
          onPress={() => onVariationPress(3)}
        />
      </>
    );
  };

  return (
    <Container>
      <AppBackground>
        <CustomHeader
          right
          left
          title={user}
          leftIcon={Icons.back}
          leftOnPress={() => navigation.goBack()}
        />
        <Content contentContainerStyle={styles.container}>
          {renderButtons()}
          <CustomSlider
            thumbIcon={Icons.diamond}
            title={Strings.txtSlider}
            sliderValue={slider}
            onSliderValueChange={(value) => {
              setSlider(value);
              if (value === 1) {
                Alert.alert(
                  AppConstants.applicationName,
                  Strings.sliderCongratulations
                );
              }
            }}
          />
        </Content>
      </AppBackground>
    </Container>
  );
};

DetailScreen.propTypes = {
  navigation: PropTypes.object
};
export default DetailScreen;
