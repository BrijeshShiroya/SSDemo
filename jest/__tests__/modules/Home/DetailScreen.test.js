import 'react-native';
import React from 'react';
import DetailScreen from '../../../../app/modules/Home/DetailScreen';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';

const initialState = {
  welcome: {
    user: 'Brijesh',
    fetching: false,
    error: null
  }
};
const mockStore = configureStore();
const store = mockStore(initialState);
// Note: test renderer must be required after react-native.

const detailScreenComponent = shallow(
  <Provider store={store}>
    <DetailScreen />
  </Provider>
);

describe('DetailScreen', () => {
  it('DetailScreen renders correctly', () => {
    expect(detailScreenComponent).toMatchSnapshot();
  });
});
