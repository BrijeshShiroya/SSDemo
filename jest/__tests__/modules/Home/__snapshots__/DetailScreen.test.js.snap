// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailScreen DetailScreen renders correctly 1`] = `ShallowWrapper {}`;
